From: jbriance <jbriance@cisco.com>
Subject: keycodes: Fix non-X11 build when use_xkbcommon=1

Since M51, ui/events/keycodes/xkb_keysym.h follows X11 path, even if
use_xkbcommon=1 is set because USE_XKBCOMMON is not defined, leading
to the following compilation error in this case:

  In file included from ../../ui/events/keycodes/keyboard_code_conversion_xkb.h:12:0,
                   from ../../ui/events/keycodes/keyboard_code_conversion_xkb.cc:5:
  ../../ui/events/keycodes/xkb_keysym.h:19:19: fatal error: X11/X.h: No such file or directory

BUG=none

Review-Url: https://codereview.chromium.org/2239833003
--

Index: ui/events/keycodes/BUILD.gn
diff --git a/ui/events/keycodes/BUILD.gn b/ui/events/keycodes/BUILD.gn
index 4f2d47c2497d321332c79d0dd4e9007b50c319c7..716c7ed17cc8049ea65b95a5062e7c6e0adc6e48 100644
--- a/ui/events/keycodes/BUILD.gn
+++ b/ui/events/keycodes/BUILD.gn
@@ -17,6 +17,10 @@ source_set("xkb") {
     "//base",
     "//ui/events:dom_keycode_converter",
   ]
+
+  if (use_xkbcommon) {
+    defines = [ "USE_XKBCOMMON" ]
+  }
 }
 
 if (use_x11 || ozone_platform_x11) {
Index: ui/events/keycodes/events_keycodes.gyp
diff --git a/ui/events/keycodes/events_keycodes.gyp b/ui/events/keycodes/events_keycodes.gyp
index b0796b4e2ead8bffe88fb01ed136814be505d967..248e961bad2ad3996ca99624b6c489c7835f6241 100644
--- a/ui/events/keycodes/events_keycodes.gyp
+++ b/ui/events/keycodes/events_keycodes.gyp
@@ -21,6 +21,13 @@
         'scoped_xkb.h',
         'xkb_keysym.h',
       ],
+      'conditions': [
+        ['use_xkbcommon==1', {
+          'defines': [
+            'USE_XKBCOMMON',
+          ],
+        }],
+      ],
     },
   ],
   'conditions': [
